### PR TITLE
Remove named dhcpd

### PR DESCRIPTION
**Pull request by jvonau**
Replacement for #3296 with [net_tweeks](https://github.com/iiab/iiab/compare/master...jvonau:net_tweeks?expand=1) #3264

### CODE REVIEW

File: roles/firmware/templates/iiab-firmware-warn.sh [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 if [ -f /tmp/.fw_modified ]; then
     echo -e "\n\e[41;1mWiFi Firmware link(s) modified, per iiab/iiab#2853: PLEASE REBOOT!\e[0m"
-    echo
-    echo -e "If you want this warning to stop, run: sudo rm /tmp/.fw_modified\n"
+    # /tmp should be auto cleaned with a reboot
```
Comment (holta) [inline roles/firmware/templates/iiab-firmware-warn.sh]: Both options / explanations should be visible:

```
If you want this warning to stop, run 'sudo rm /tmp/.fw_modified' or reboot.
```

Operators get very unhappy when loud warnings appear without explanation, or worse explanation(s) hidden from them.

(And even more so if firmware-install-on-each-boot glitch were to occur at anytime in future.)

File: roles/firmware/tasks/main.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 - name: Install firmware (for RPi internal WiFi)
   include_tasks: install.yml
-  #when: firmware_installed is undefined
+  when: firmware_installed is undefined
```
Comment (holta) [inline roles/firmware/tasks/main.yml:21]: Debatable.

Downloading the latest firmware every time might in fact be desired and preferable.

Comment (jvonau) [reply roles/firmware/tasks/main.yml:21]: Both iiab-network and admin-console's 'services' but network goes along for the ride, are expected to work offline or when the internet is not available. Not sure what effect of SKIP_ROLE_ON_ERROR might have changed things.

File: roles/firmware/templates/iiab-firmware-warn.sh (hunk 1 above)
Comment (jvonau) [reply roles/firmware/templates/iiab-firmware-warn.sh]: No the correct fix is to reboot, period the firmware was replaced. As you know /tmp/.fw_modified doesn't survive a reboot. removing that file just suppress the warning from other new logins so what is gained when the only correct answer is to reboot, this is not a suggestion. This warning should really be extended to the desktops, perhaps display in box.lan/home #3309 and not just ssh sessions.

Comment (holta) [reply roles/firmware/templates/iiab-firmware-warn.sh]: The explanation should explain clearly, one way or another.  In that 2nd line (it's more white not red) that serves an important purpose (UX transparency).  Example:

```
If you want this warning to stop, reboot to remove /tmp/.fw_modified
```

Comment (jvonau) [reply roles/firmware/templates/iiab-firmware-warn.sh]: Look If you wanted the second line to say 'sudo rm /tmp/.fw_modified' or reboot' or whatever you can change it in the tree now and I'll rebase, just keep the changes small given the largeness of this PR that I would like to get in quickly.

Comment (holta) [reply roles/firmware/templates/iiab-firmware-warn.sh]: Done:

https://github.com/iiab/iiab/commit/a2cbfc45d6f558e493272d4d8fb93c91b2a4275e

**Comment by holta** [comment]
What aspects of this PR are most important to test?

Smoke Test on Ubuntu Server 22.04 VM: http://sprunge.us/NHMUFH?en

**Comment by jvonau** [comment]
Nothing else really, for the most part this is disabling optional features which by default would not be used unless selected. New feature is local_vars validation should these optional features be engaged by selecting install True to advise of the now unavailable option.

**Comment by jvonau** [comment]
Notes and explanation

1113 =IIAB==========================================================================
1114 COMMAND: /usr/bin/sudo journalctl -b 0 -u dnsmasq    # dnsmasq log
1115 
1116 Jul 17 21:26:13 box systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
1117 Jul 17 21:26:13 box dnsmasq[9156]: dnsmasq: failed to create listening socket for port 53: Address already in use
1118 Jul 17 21:26:13 box dnsmasq[9156]: failed to create listening socket for port 53: Address already in use
1119 Jul 17 21:26:13 box dnsmasq[9156]: FAILED to start up
1120 Jul 17 21:26:13 box systemd[1]: dnsmasq.service: Control process exited, code=exited, status=2/INVALIDARGUMENT
1121 Jul 17 21:26:13 box systemd[1]: dnsmasq.service: Failed with result 'exit-code'.
1122 Jul 17 21:26:13 box systemd[1]: Failed to start dnsmasq - A lightweight DHCP and caching DNS server.

This is the stock apt install of dnsmasq with issue #1306

1123 Jul 17 21:26:17 box systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
1124 Jul 17 21:26:17 box dnsmasq[9493]: started, version 2.86 cachesize 150
1125 Jul 17 21:26:17 box dnsmasq[9493]: DNS service limited to local subnets
1126 Jul 17 21:26:17 box dnsmasq[9493]: compile time options: IPv6 GNU-getopt DBus no-UBus i18n IDN2 DHCP DHCPv6 no-Lua TFTP conntrack ipset auth cryptohash DNSSEC loop-detect inotify dumpfile
1127 Jul 17 21:26:17 box dnsmasq[9493]: reading /etc/resolv.conf
1128 Jul 17 21:26:17 box dnsmasq[9493]: using nameserver 127.0.0.53#53
1129 Jul 17 21:26:17 box dnsmasq[9493]: read /etc/hosts - 5 addresses
1130 Jul 17 21:26:17 box systemd[1]: Started dnsmasq - A lightweight DHCP and caching DNS server.

Now the [fix](https://github.com/iiab/iiab/blob/master/roles/network/templates/network/dnsmasq-iiab) delivered in [dnsmasq.yml](https://github.com/iiab/iiab/blob/master/roles/network/tasks/dnsmasq.yml)

1131 Jul 17 21:51:26 box systemd[1]: Stopping dnsmasq - A lightweight DHCP and caching DNS server...
1132 Jul 17 21:51:26 box dnsmasq[9493]: exiting on receipt of SIGTERM
1133 Jul 17 21:51:26 box systemd[1]: dnsmasq.service: Deactivated successfully.
1134 Jul 17 21:51:26 box systemd[1]: Stopped dnsmasq - A lightweight DHCP and caching DNS server.

captive portal

1135 Jul 17 21:51:27 box systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
1136 Jul 17 21:51:27 box dnsmasq[41230]: started, version 2.86 cachesize 150
1137 Jul 17 21:51:27 box dnsmasq[41230]: DNS service limited to local subnets
1138 Jul 17 21:51:27 box dnsmasq[41230]: compile time options: IPv6 GNU-getopt DBus no-UBus i18n IDN2 DHCP DHCPv6 no-Lua TFTP conntrack ipset auth cryptohash DNSSEC loop-detect inotify dumpfile
1139 Jul 17 21:51:27 box dnsmasq[41230]: reading /etc/resolv.conf
1140 Jul 17 21:51:27 box dnsmasq[41230]: using nameserver 127.0.0.53#53
1141 Jul 17 21:51:27 box dnsmasq[41230]: read /etc/hosts - 5 addresses
1142 Jul 17 21:51:27 box systemd[1]: Started dnsmasq - A lightweight DHCP and caching DNS server.

Start of the network role being run.

1143 Jul 17 21:52:34 box dnsmasq[41230]: reading /etc/resolv.conf
1144 Jul 17 21:52:34 box dnsmasq[41230]: using nameserver 127.0.0.53#53
1145 Jul 17 21:52:34 box dnsmasq[41230]: reading /etc/resolv.conf
1146 Jul 17 21:52:34 box dnsmasq[41230]: using nameserver 127.0.0.53#53
1147 Jul 17 21:52:36 box dnsmasq[41230]: reading /etc/resolv.conf
1148 Jul 17 21:52:36 box dnsmasq[41230]: using nameserver 127.0.0.53#53
1149 Jul 17 21:52:36 box dnsmasq[41230]: reading /etc/resolv.conf
1150 Jul 17 21:52:36 box dnsmasq[41230]: using nameserver 127.0.0.53#53
1151 Jul 17 21:52:36 box dnsmasq[41230]: reading /etc/resolv.conf
1152 Jul 17 21:52:36 box dnsmasq[41230]: using nameserver 127.0.0.53#53
1153 Jul 17 21:52:36 box dnsmasq[41230]: reading /etc/resolv.conf
1154 Jul 17 21:52:36 box dnsmasq[41230]: using nameserver 127.0.0.53#53
1155 Jul 17 21:52:36 box dnsmasq[41230]: reading /etc/resolv.conf
1156 Jul 17 21:52:36 box dnsmasq[41230]: using nameserver 127.0.0.53#53
1157 Jul 17 21:52:36 box dnsmasq[41230]: reading /etc/resolv.conf
1158 Jul 17 21:52:36 box dnsmasq[41230]: using nameserver 127.0.0.53#53
1159 Jul 17 21:52:36 box dnsmasq[41230]: reading /etc/resolv.conf
1160 Jul 17 21:52:36 box dnsmasq[41230]: using nameserver 127.0.0.53#53

Restarting of the network.

1161 Jul 17 21:52:47 box systemd[1]: Stopping dnsmasq - A lightweight DHCP and caching DNS server...
1162 Jul 17 21:52:47 box dnsmasq[41230]: exiting on receipt of SIGTERM
1163 Jul 17 21:52:47 box systemd[1]: dnsmasq.service: Deactivated successfully.
1164 Jul 17 21:52:47 box systemd[1]: Stopped dnsmasq - A lightweight DHCP and caching DNS server.
1165 Jul 17 21:52:47 box systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
1166 Jul 17 21:52:47 box dnsmasq[45745]: started, version 2.86 cachesize 150
1167 Jul 17 21:52:47 box dnsmasq[45745]: DNS service limited to local subnets
1168 Jul 17 21:52:47 box dnsmasq[45745]: compile time options: IPv6 GNU-getopt DBus no-UBus i18n IDN2 DHCP DHCPv6 no-Lua TFTP conntrack ipset auth cryptohash DNSSEC loop-detect inotify dumpfile
1169 Jul 17 21:52:47 box dnsmasq[45745]: reading /etc/resolv.conf
1170 Jul 17 21:52:47 box dnsmasq[45745]: using nameserver 127.0.0.53#53
1171 Jul 17 21:52:47 box dnsmasq[45745]: read /etc/hosts - 5 addresses
1172 Jul 17 21:52:47 box systemd[1]: Started dnsmasq - A lightweight DHCP and caching DNS server.

final restart in network restart.yml 
Notes on dnsmasq with br0 [restarting](https://github.com/iiab/iiab/pull/1765) with [networkd-dispatcher](https://github.com/iiab/iiab/pull/1779)

**Comment by jvonau** [comment]
df trigger
```
1878 Jul 17 22:07:42 vm18 systemd[1]: proc-sys-fs-binfmt_misc.automount: Got automount request for /proc/sys/fs/binfmt_misc, triggered by 55711 (df)
1879 Jul 17 22:07:42 vm18 systemd[1]: Mounting Arbitrary Executable File Formats File System...
1880 Jul 17 22:07:42 vm18 systemd[1]: Mounted Arbitrary Executable File Formats File System.
```

**Pull request merged**